homeassistant/components/agent_dvr/camera.py

```diff
@@ -0,0 +1,263 @@
+        self._enabled_default = enabled_default
+        super().__init__(device_info)
+
+    @property
+    def device_info(self):
+        """Return the device info for adding the entity to the agent object."""
+        return {
+            "identifiers": {(AGENT_DOMAIN, self._unique_id)},
+            "name": self._name,
+            "manufacturer": "Agent",
+            "model": "Camera",
+            "sw_version": self.device.client.version,
+        }
+
+    async def async_update(self):
+        """Update our state from the Agent API."""
+        try:
+            await self.device.update()
+            if self._removed:
+                _LOGGER.error("%s reacquired", self._name)
+            self._removed = False
+        except AgentError:
+            if self.device.client.is_available:  # server still available - camera error
+                if not self._removed:
+                    _LOGGER.error("%s lost", self._name)
+                    self._removed = True
+
+    @property
+    def state_attributes(self):
+        """Return the Agent DVR camera state attributes."""
+        return {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            "editable": False,
+            "enabled": self.enabled,
+            "connected": self.connected,
+            "detected": self.is_detected,
+            "alerted": self.is_alerted,
+            "recording": self.is_recording,
+            "has_ptz": self.device.has_ptz,
+            "motion_detection_enabled": self.device.detector_active,
+            "alerts_enabled": self.device.alerts_active,
+        }
+
+    @property
+    def is_recording(self) -> bool:
+        """Return whether the monitor is recording."""
+        return self.device.recording
+
+    @property
+    def is_alerted(self) -> bool:
+        """Return whether the monitor has alerted."""
+        return self.device.alerted
+
+    @property
+    def is_detected(self) -> bool:
+        """Return whether the monitor has alerted."""
+        return self.device.detected
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.device.client.is_available
+
+    @property
+    def connected(self) -> bool:
+        """Return True if entity is connected."""
+        return self.device.connected
+
+    @property
+    def enabled(self) -> bool:
+        """Return True if entity is enabled."""
+        return self.device.online
+
+    @property
+    def supported_features(self) -> int:
+        """Return supported features."""
+        return SUPPORT_ON_OFF | SUPPORT_STREAM
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if on."""
+        return self.device.online
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        if self.enabled:
+            return "mdi:camcorder"
+        return "mdi:camcorder-off"
+
+    @property
+    def state(self):
```

Comment:
The `state` property should not be overridden. The base camera entity is responsible for it.

Reply:
I don't understand - how will the base camera entity know if Agent is recording the camera or if the Camera is online? Both these states come from Agent.

Reply:
you know what i'll just remove it, i'm done with it

Reply:
Implement `is_recording` and `is_streaming` properties instead.

https://github.com/home-assistant/core/blob/1b3f9254bc5b392c29a3219bb1b82d127036d18c/homeassistant/components/camera/__init__.py#L341-L344

Reply:
Please implement a property `is_streaming` too.

Reply:
I added an is_streaming:
`    @property
    def is_streaming(self) -> bool:
        """Return whether the monitor is online."""
        return self.device.connected`

.. and it broke the integration:

`2020-05-08 03:07:16 ERROR (MainThread) [homeassistant.components.camera] Error while setting up agent_dvr platform for camera
Traceback (most recent call last):
  File "/workspaces/home-assistant/homeassistant/helpers/entity_platform.py", line 178, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.8/asyncio/tasks.py", line 483, in wait_for
    return fut.result()
  File "/workspaces/home-assistant/homeassistant/components/agent_dvr/camera.py", line 57, in async_setup_entry
    camera = AgentCamera(device)
  File "/workspaces/home-assistant/homeassistant/components/agent_dvr/camera.py", line 84, in __init__
    super().__init__(device_info)
  File "/workspaces/home-assistant/homeassistant/components/mjpeg/camera.py", line 92, in __init__
    super().__init__()
  File "/workspaces/home-assistant/homeassistant/components/camera/__init__.py", line 320, in __init__
    self.is_streaming = False`